File: Python/Lab4/TTriad.py
from abc import abstractmethod


class TTriad:
    def __init__(self, number1, number2, number3):
        self.__number1 = number1
        self.__number2 = number2
        self.__number3 = number3

    @property
    def Number1(self):
        return self.__number1

    @Number1.setter
    def Number1(self, number1):
        self.__number1 = number1

    @property
    def Number2(self):
        return self.__number2

    @Number2.setter
    def Number2(self, number2):
        self.__number2 = number2

    @property
    def Number3(self):
        return self.__number3

    @Number3.setter
    def Number3(self, number3):
        self.__number3 = number3


    @abstractmethod
    def IncreaseNumber1(self, n):
        pass

    @abstractmethod
    def DecreaseNumber1(self, n):
        pass

    @abstractmethod
    def IncreaseNumber2(self, n):
        pass

    @abstractmethod
    def DecreaseNumber2(self, n):
        pass

    @abstractmethod
    def IncreaseNumber3(self, n):
        pass

    @abstractmethod
    def DecreaseNumber3(self, n):
        pass

File: C#/Lab4/TTriad.cs
namespace lab4
{
    public abstract class TTriad
    {
        public int Number1 { get; set; }
        public int Number2 { get; set; }
        public int Number3 { get; set; }
        
        protected TTriad(int n1, int n2, int n3)
        {
            Number1 = n1;
            Number2 = n2;
            Number3 = n3;
        }
        
        public abstract void IncreaseNumber1(int n);
        public abstract void DecreaseNumber1(int n);
        public abstract void IncreaseNumber2(int n);
        public abstract void DecreaseNumber2(int n);
        public abstract void IncreaseNumber3(int n);
        public abstract void DecreaseNumber3(int n);
    }
}